geode-core/src/main/java/org/apache/geode/management/internal/api/LocatorClusterManagementService.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.geode.management.internal.api;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.logging.log4j.Logger;
+
+import org.apache.geode.cache.configuration.CacheConfig;
+import org.apache.geode.cache.configuration.ClusterCacheElement;
+import org.apache.geode.cache.execute.Function;
+import org.apache.geode.cache.execute.ResultCollector;
+import org.apache.geode.distributed.ConfigurationPersistenceService;
+import org.apache.geode.distributed.DistributedMember;
+import org.apache.geode.internal.cache.InternalCache;
+import org.apache.geode.internal.logging.LogService;
+import org.apache.geode.management.internal.cli.CliUtil;
+import org.apache.geode.management.internal.cli.functions.CliFunctionResult;
+import org.apache.geode.management.internal.cli.functions.UpdateCacheFunction;
+import org.apache.geode.management.internal.exceptions.EntityExistsException;
+import org.apache.geode.management.internal.exceptions.NoMembersException;
+
+public class LocatorClusterManagementService implements ClusterManagementService {
+  private static Logger logger = LogService.getLogger();
+  private InternalCache cache;
+  private ConfigurationPersistenceService persistenceService;
+
+  public LocatorClusterManagementService(InternalCache cache,
+      ConfigurationPersistenceService persistenceService) {
+    this.cache = cache;
+    this.persistenceService = persistenceService;
+  }
+
+  @Override
+  public APIResult createCacheElement(ClusterCacheElement config) {
+    APIResult result = new APIResult();
+    CacheConfig cacheConfig;
+    String group = "cluster";
+
+    // assuming config is validated at this point
+    if (persistenceService != null) {
```

Comment:
No, you can have a region without persistence enabled, apparently.

Reply:
I thought this persistence service was relating to the cluster configuration mechanism rather than "persistence" in the sense of a region writing to disk. I thought ClusterConfig, when enabled, would persist locally by default.

Reply:
Sorry, that's what I meant -- cluster config enabled versus not.

Reply:
Could this then maybe just have a pass-through persistence mechanism? Makes more sense than having to null check

Reply:
You still need to have the persistenceService available here in order to grab the latest cluster config object. Figure if you need it, you might as well use its nullability to check if persistence is enabled.